### ISSUE TITLE
[2.8.8] Duplicate services after reload

### ISSUE THREAD

**Issue by itsul** [centreon/centreon#5321]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**Centreon Web version**: 
centreon-2.8.8-4.el6
**Centreon Engine version**: 
centreon-engine-1.7.2-3.el6
**Centreon Broker version**: 
centreon-broker-3.0.7-1.el6
**OS**: 
CentOS release 6.9
**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical

**Steps to reproduce the issue:**
1. Go to Configuration > Hosts 
2. Delete a Host (Host1)
3. Copy a similar Host (Same Checks/Templates)
4. Rename the copied Host like the one you deleted (Host2_1 to Host1)
5. Go to Configuration  >  Pollers  >  Export configuration
6. !Reload! the configuration

![selection_123](https://user-images.githubusercontent.com/12328177/26871231-be35f164-4b72-11e7-9408-2be4505ab09b.png)

**Describe the results you received:**
Go To Monitoring  >  Status Details  >  Services | Every Service is now Duplicated, if you do the same procedure again your Services get tripled and so on
![selection_122](https://user-images.githubusercontent.com/12328177/26870955-f2fb7870-4b71-11e7-80a4-614aeee1d13a.png)

**Additional information you think important (e.g. issue happens only occasionally):**
Only happens if you !Reload! the engine, a Restart doesn't cause this issue and even "fix" it

**Comment by jben94** [comment]
Hi,

The problem is when removing and creating a service with the same name and description.

More precisely, the problem lies in the centreon-engine.
When the difference between the old configuration and the new one.
The key using is Name + Description.
Centeron engine believes that the service has just been changed and then that is new.

Waiting for a better solution, I changed the file "difference.hh".

```
          else if (**first1 != **first2) {
            *modif++ = *first2++;
            ++first1;
          }
```

to 

```
          else if (*first1 != *first2) {
            *del++ = *first1++;
            *add++ = *first2++;
          }
```

This solution causes a change to the "Pending" state instead of the previous state.
Not knowing the C ++, I leave the developer to find a better solution ;) @ganoze
